### DISCUSSION TITLE
Automatic PR linking behavior and exceptions?

### DISCUSSION THREAD

**Discussion by boneskull** [community/community#23712] [New to GitHub]
<p>In the <a href="https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue" rel="noopener nofollow ugc">docs</a>, it mentions (at the bottom):</p>
<blockquote>
<p>You can also use closing keywords in a commit message. The issue will be closed when you merge the commit into the default branch, but the pull request that contains the commit will not be listed as a linked pull request.</p>
</blockquote>
<ol>
<li>Why is a <em>commit message</em> an exception?</li>
<li>The issue <em>title</em> also seems to be an exception to this rule.</li>
</ol>
<p>I tend to write commit messages like this:</p>
<pre><code class="lang-auto">fixes the broken thingy; closes #1234
</code></pre>
<p>When I open a PR for my branch, the title of the PR is automatically populated with the first line of my commit message.</p>
<p>But neither the title nor the commit actually performs the link, even though issue <span class="hashtag">#1234</span> will be closed upon merge.</p>
<p>I imagine this can be worked around via a GitHub action, but I’m curious why the automatic linking behavior is only enabled for the PR description?</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hey <a class="mention" href="/boneskull">@boneskull</a> :wave:</p>
<p>Thanks so much for writing in! I’ve read your post a few times now and I think I may be somewhat confused.</p>
<p>I understand that titles don’t include the auto-linking behavior that you see in a commit message, or PR.</p>
<p>Though your questions of “why,” don’t these links work the way you suggest don’t have much of a substantive answer.</p>
<p>What I would recommend, is that you reform this post to our Product team, via this form:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://support-assets.github.com/assets/favicon-3dcd2c977a91b2e3f396b300c5ff0d572295ca632ddb7ba1adcacc1ea68dd5c0.ico" class="site-icon" width="16" height="16">
      <a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">support.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://support.github.com/contact/feedback" target="_blank" rel="noopener">Share feedback - GitHub Support</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…since it doesn’t sound like anything unexpected is happening.</p>
<p>That form sends your feedback directly to our PM team for review and consideration.</p>
<p>If there’s something that I’ve misunderstood, please let me know and we can address it here. :bow:</p>
